### PR TITLE
Add engineering teams description

### LINKED ISSUE #72: Describe teams and process
For an outside person it's not very clear what Data Intelligence, Machine Learning and Applications teams are doing and how they communicate with each other. (for other teams it's more or less clear because of the name, but not how they are connected)
In my opinion it's worth to explain the process of development here. Some graph like this:

```
Data Retrieval => Language Analysis => Machine Learning 
                                                        =>  Applications => Infrastructure
                          Product
```

And some description like:

Data Retrieval - retrieve and store source code from one project to all of the world's public code. Build such projects as go-git, rovers, borges, siva
Language Analysis - language detection on a large scale and build querying, features extraction, collects statistics for any programming language that enables code analysis for all open source code and enterprise. Build such projects as envy, babelfish
...
Applications - applies whole stack and expertise to solve real word problems as code duplication, smart diffs, search for repositories, projects evolving reports
Infrastructure - runs kubernetes cluster on bare metal and AWS to allow process millions of code lines every minute

P.S. Anything above can be incorrect and just my fantasy, because I don't really know how everything works yet.

### PR DESCRIPTION
**Pull request by tsolakoua**
Fix #72 
I used information and images from the job descriptions as well as relevant docs we have. 

The engineering team knows better how each team interacts with each other, so any feedback is appreciated.

### CODE REVIEW

**Comment by dpordomingo** [comment]
Team leads might be interested
@rporres @bzz @vmarkovtsev @ajnavarro (plus @src-d/language-analysis mates)

File: talent/engineering-teams-descr.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+This document describes the engineering teams at source{d} and how they interact with each other.
+
+### Machine Learning Team
+
+- The Machine Learning team is focused on providing intelligent insights into nearly all the world's Open Source code. This includes typical Big Data analysis as well as solving sophisticated Machine Learning problems. The former is conducted on Apache Spark clusters, the latter leverages extensive GPGPU acceleration on custom hardware using deep learning techniques. Often ML problems we face are unsupervised and require novel research in the domain of ML on source code (MLoSC).
```
Comment (dpordomingo) [inline talent/engineering-teams-descr.md]: I saw @eiso and @vmarkovtsev are writing `MLonCode` instead of ~~`MLoSC`~~
They could confirm it

File: talent/engineering-teams-descr.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+### Applications Team
+
+- The Applications team applies the whole stack and expertise to solve real word problems as code duplication, smart diffs, search for repositories, projects evolving reports. This is done by implementing CLI/Web applications on top of the MLoSC research. Simple and intuitive applications are written mainly in Go and Scala, based on developer tooling prototypes.
```
Comment (dpordomingo) [inline talent/engineering-teams-descr.md]: review `MLonCode`

Comment (dpordomingo) [inline talent/engineering-teams-descr.md]: I'd say that Apps build apps on top of other teams main products, not only `MLoSC`; that's because we also create apps on top of Bblfsh, Engine or GitBase.

File: talent/engineering-teams-descr.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+### Data Retrieval Team 
+
+- The Data Retrieval team is developing source{d}'s high-level Engine for running scalable data retrieval pipelines that process and manipulate any number of code repositories for source code analysis. Written mostly in Scala, it aims to be robust, friendly and flexible: it is built on top of Apache Spark, accessible both via Scala and Python Spark APIs, and capable of running on large-scale distributed clusters over petabytes of data.
```
Comment (dpordomingo) [inline talent/engineering-teams-descr.md]: go-git and git-base are not relevant to be mentioned?

File: talent/engineering-teams-descr.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+- Our cluster, with close to 2000 threads, more than 11TB of RAM and 1PB of storage, is backed by CoreOS and Kubernetes. It has two main goals: on the one hand, it stores all the available public code in a HDFS cluster and on the other hand, it runs intensive computation jobs over the stored data on top of Spark.
+
+- Besides the main points above, the infrastructure team is also in charge of several other important services, such as databases, queues, CI, monitorization, logging, etc.
+
+### Communication 
+
+- The ```Applications``` team is responsible for building scalable developer tools on top of the research the ```Machine Learning``` team is doing. The ```Data Retrieval``` team retrieves and processes public source code in order to build the data pipelines for the Machine Learning team’s models. For that, BabelFish from ```Language Analysis``` team is used in order to generate the datasets as Universal Abstract Syntax tree to be analyzed directly or given as input data to train the ML models. All the necessary services and computational clusters for every team, are supported by the Infrastructure team.
```
Comment (dpordomingo) [inline talent/engineering-teams-descr.md]: Imo this paragraph looks weird. Some parts are already mentioned in each team description (ie Apps team); in other cases, there is new data here that would be better in the team description (ie, DR team, retrieving and processing all the OS over the world)

Imo the relation between teams can be understood with the team descriptions + attached graph, so it would not be necessary an extra (redundant) paragraph.

File: talent/engineering-teams-descr.md (hunk 1 above)
Comment (tsolakoua) [reply talent/engineering-teams-descr.md]: In the last updates in the job descriptions around 3 weeks ago, they were written as MLoSC.

File: talent/engineering-teams-descr.md (hunk 2 above)
Comment (tsolakoua) [reply talent/engineering-teams-descr.md]: That's true, thanks @dpordomingo

File: talent/engineering-teams-descr.md (hunk 3 above)
Comment (tsolakoua) [reply talent/engineering-teams-descr.md]: @ajnavarro I need your input here, as I'm not sure either. I guess they should be relevant.

File: talent/engineering-teams-descr.md (hunk 4 above)
Comment (tsolakoua) [reply talent/engineering-teams-descr.md]: The idea of a graph sounds great. I would still keep the text though along with the graph I think. @dpordomingo do you use any specific tools to make graphs for documents here or I can use whatever is convenient for me? Let me know if you have any suggestions.

File: talent/engineering-teams-descr.md (hunk 1 above)
Comment (dpordomingo) [reply talent/engineering-teams-descr.md]: :)

File: talent/engineering-teams-descr.md (hunk 4 above)
Comment (tsolakoua) [reply talent/engineering-teams-descr.md]: ....after discussing with @dpordomingo on slack, maybe @bzz you have any graph/drawing tools suggest so we can be consistent across the repos?

File: talent/engineering-teams-descr.md (hunk 1 above)
Comment (tsolakoua) [reply talent/engineering-teams-descr.md]: Thanks @dpordomingo for asking on general-slack!

**Comment by tsolakoua** [comment]
The diagram added in the last PR is https://docs.google.com/document/d/1ktYN6inIgCMJU4hpgzQfLW399U_jXPeqvj746tbNzq4/edit?usp=sharing. Anyone is free to make any changes in the diagram or comments in case I am missing something

File: talent/engineering-teams-descr.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+This document describes the engineering teams at source{d} and how they interact with each other.
+
+### Machine Learning Team
+
+- The Machine Learning team is focused on providing intelligent insights into nearly all the world's Open Source code. This includes typical Big Data analysis as well as solving sophisticated Machine Learning problems. The former is conducted on Apache Spark clusters, the latter leverages extensive GPGPU acceleration on custom hardware using deep learning techniques. Often ML problems we face are unsupervised and require novel research in the domain of ML on source code (MLonCode).
+
+### Applications Team
+
+- The Applications team applies the whole stack and expertise to solve real word problems as code duplication, smart diffs, search for repositories, projects evolving reports. This is done by implementing CLI/Web applications on top of the MLonCode research as well as the other teams main tools, like BabelFish, Engine, GitBase etc. Simple and intuitive applications are written mainly in Go and Scala, based on developer tooling prototypes.
```
Comment (bzz) [inline talent/engineering-teams-descr.md]: Sorry for confusion, but as it's the first time I hear about it - wondering, what are the  `developer tooling prototypes`?

Comment (tsolakoua) [reply talent/engineering-teams-descr.md]: @bzz I am looking at the Applications team (SF and Madrid) as a whole and one team as they actually are as far as I know. Taking into account their descriptions https://github.com/src-d/guide/blob/master/talent/job-descriptions/offers/applications-sf.md and https://github.com/src-d/guide/blob/master/talent/job-descriptions/offers/applications.md
it seems that the Applications team is building developer tools based on ML/LA work as well as on the prototypes built by SF. As far as I understand, at the end of the day, what Applications team is doing is building Developer Tools. Correct me if I am wrong and feel free to provide the actual description as you and your team know better.

Comment (bzz) [reply talent/engineering-teams-descr.md]: Thank you for explaining.

>  one team as they actually are as far as I know
Yes, that's something that has been discussed so far. Was confused by two entities on the diagram though.

> as well as on the prototypes built by SF

I think that might not be something we have decided up on, and thus I would suggest removing this, to avoid the confusion.

Just `Simple and intuitive applications are written mainly in Go and Scala.` should be enough.

File: talent/engineering-teams-description.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+- Babelfish powers the cutting-edge research that our Machine Learning team does on their ambitious mission to provide Machine Learning on Source Code. Our tools also are being used by a community of developers, researchers, and companies across the world in order to do vanguard research and build the next generation of developer tools.
+
+### Infrastructure Team
+
+- The Infrastructure team focuses on managing and growing our bare metal computation cluster. This cluster allows us to run large scale computational algorithms and store the required information for those computations.
+
+- Our cluster, with close to 2000 threads, more than 11TB of RAM and 1PB of storage, is backed by CoreOS and Kubernetes. It has two main goals: on the one hand, it stores all the available public code in a HDFS cluster and on the other hand, it runs intensive computation jobs over the stored data on top of Spark.
```
Comment (rporres) [inline talent/engineering-teams-description.md]: This is getting old and soon will not be true... I'd just remove it.

Comment (tsolakoua) [reply talent/engineering-teams-description.md]: @rporres could you provide some info to update it properly? I'm not sure where I should look for. This text is also in the job description on the website so we'll have to change it there as well

Comment (rporres) [reply talent/engineering-teams-description.md]: I'd just remove the cluster information paragraph in the team description... Whenever we have something decided and stable I can update it.

As for the job description, I will take a look to rephrase it to cover more ground without being so specific.

File: talent/engineering-teams-description.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+This document describes the engineering teams at source{d} and how they interact with each other.
+
+### Machine Learning Team
+
+- The Machine Learning team is focused on providing intelligent insights into nearly all the world's Open Source code. This includes typical Big Data analysis as well as solving sophisticated Machine Learning problems. The former is conducted on Apache Spark clusters, the latter leverages extensive GPGPU acceleration on custom hardware using deep learning techniques. Often ML problems we face are unsupervised and require novel research in the domain of ML on source code (MLonCode).
```
Comment (eiso) [inline talent/engineering-teams-description.md]: I would mention deep learning here

File: talent/engineering-teams-description.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+### Applications Team
+
+- The Applications team applies the whole stack and expertise to solve real word problems as code duplication, smart diffs, search for repositories, projects evolving reports. This is done by implementing CLI/Web applications on top of the MLonCode research as well as the other teams main tools, like BabelFish, Engine, GitBase etc. Simple and intuitive applications are written mainly in Go and Scala.
```
Comment (eiso) [inline talent/engineering-teams-description.md]: s/problems as/problems such as/

I also don't think this list is representative of the work applications will be doing. Since we won't tackle projects like search. See the new slide (4) added to the tech deck

File: talent/engineering-teams-description.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+### Data Retrieval Team 
+
```
Comment (eiso) [inline talent/engineering-teams-description.md]: I would mention GitBase here.
